### PR TITLE
The FloorSlider can now have custom values.

### PR DESCRIPTION
**Pull request by pvalsecc**
Added an optional config entry "floors" that can get the list of floors to
show.

### CODE REVIEW

File: core/src/script/CGXP/widgets/FloorSlider.js [hunk 1]
```diff
@@ -153,6 +158,61 @@ cgxp.FloorSlider = Ext.extend(Ext.Window, {
                 this.anchorPosition, this.anchorOffsets]);
     },
 
+    /** private: method[fixFloors]
+    */
+    fixFloors: function(config) {
```
Comment (sbrunner) [inline core/src/script/CGXP/widgets/FloorSlider.js]: Why `fixFloors` ? something like `generateFloorMap` isn't better?

File: core/src/script/CGXP/widgets/FloorSlider.js [hunk 2]
```diff
@@ -153,6 +158,61 @@ cgxp.FloorSlider = Ext.extend(Ext.Window, {
...
+        if (this.maxIsSky && floor === null) {
+            return this.maxValue;
+        }
+        if (this.floors) {
+            for (var pos = 0; pos < this.floors.length; ++pos) {
+                if (this.floors[pos] == floor) {
+                    return pos;
+                }
+            }
+            return this.value;
```
Comment (sbrunner) [inline core/src/script/CGXP/widgets/FloorSlider.js]: Why this `return`?

File: core/src/script/CGXP/widgets/FloorSlider.js [hunk 3]
```diff
@@ -153,6 +158,61 @@ cgxp.FloorSlider = Ext.extend(Ext.Window, {
...
+                    floorMap[i] = this.skyText;
+                } else {
+                    floorMap[i] = i.toString();
+                }
+            }
+        } else {
+            //floors provided => compute min/max
+            config.minValue = 0;
+            config.maxValue = config.floors.length - 1;
+            for(var i = 0; i <= config.maxValue; ++i) {
```
Comment (sbrunner) [inline core/src/script/CGXP/widgets/FloorSlider.js]: i already defined :-)

**Comment by sbrunner** [comment]
Just some small things otherwise that looks good :-)

File: core/src/script/CGXP/widgets/FloorSlider.js (hunk 2 above)
Comment (pvalsecc) [inline core/src/script/CGXP/widgets/FloorSlider.js]: OK, I'll throw an exception, instead

**Comment by sbrunner** [comment]
Looks good, merge :-)

**Pull request merged**